### ISSUE TITLE
E684: list index out of range: -1

### ISSUE THREAD

**Issue by kassio** [junegunn/vim-plug#328]
I upgraded to last nvim version, and now I'm getting this error on `PlugUpdate`:

```
Error detected while processing function <SNR>3_job_handler..<SNR>3_tick..<SNR>3_progress_opt..<SNR>3_git_version_requirement:
line    1:
E684: list index out of range: -1
Error detected while processing function <SNR>3_job_handler..<SNR>3_tick..<SNR>3_progress_opt..<SNR>3_git_version_requirement:
line    1:
E116: Invalid arguments for function split(split(s:system('git --version'))[-1], '\.'), 'str2nr(v:val)'))
Error detected while processing function <SNR>3_job_handler..<SNR>3_tick..<SNR>3_progress_opt..<SNR>3_git_version_requirement:
line    1:
E116: Invalid arguments for function map(split(split(s:system('git --version'))[-1], '\.'), 'str2nr(v:val)'))
Error detected while processing function <SNR>3_job_handler..<SNR>3_tick..<SNR>3_progress_opt..<SNR>3_git_version_requirement:
line    1:
E116: Invalid arguments for function get(s:, 'git_version', map(split(split(s:system('git --version'))[-1], '\.'), 'str2nr(v:val)'))
Error detected while processing function <SNR>3_job_handler..<SNR>3_tick..<SNR>3_progress_opt..<SNR>3_git_version_requirement:
line    1:
E15: Invalid expression: get(s:, 'git_version', map(split(split(s:system('git --version'))[-1], '\.'), 'str2nr(v:val)'))
```

**nvim version**
```
:version
NVIM v0.1.0-114-g5da7ea9 (compiled Nov 25 2015 14:56:22)
Commit: 5da7ea98cac53549dbbbbfafd7fcda3d93e3f99c
```

**Comment by kassio** [comment]
It happened only once, I'll close it for now, if happens again I'll reopen.

**Comment by starcraftman** [comment]
@kassio Ok well if it happens again make a new ticket. I don't use the nvim version too often.

**Comment by junegunn** [comment]
I also ran into this while fixing the output of nvim installer. In my case, nvim crashed. Not easily reproducible though. /cc @tarruda

```
Error detected while processing function <SNR>2_job_handler..<SNR>2_tick..<SNR>2_progress_opt..<SNR>2_git_version_requirement:
line    1:
E684: list index out of range: -1
Press ENTER or type command to continueAssertion failed: (!queue->parent), function queue_remove, file /tmp/neovim20151119-90430-1vlo7qb/neovim-0.1.0/src/nvim/event/queue.c, line 168.
Abort trap: 6
```

**Comment by kaldown** [comment]
nvim 0.3.4
```
Error detected while processing function <SNR>3_job_handler:
line    8:
E121: Undefined variable: self
```
for any kind of a plugin

**Comment by kaldown** [comment]
fxd by upgrading 0.9.1 to 0.10.0
